### PR TITLE
Feature/search and category

### LINKED ISSUE #15: User Story | Browse Items
- [ ] As a user, I should be able to browse & search items available to borrow and to keep

### LINKED ISSUE #17: User Story | Search
- [ ] As a user, I should be able to search available items

### PR DESCRIPTION
**Pull request by Abumpus1**
## What was changed:
- Separates search bar into component
- Adds functionality to sort by category and search by name
- Adds E2E tests to Landing Page
- Swaps colors of cards to be based on category instead of random

## Explain the reason for changes:
- Allows user to filter results, tested in browser and with newly added E2E tests

## Link to issue:
closes #17 
closes #15

### CODE REVIEW

File: cypress/e2e/1-getting-started/landingPage.cy.js [hunk 1]
```diff
@@ -1,16 +1,111 @@
...
+    cy.get(".category-section").children().eq(1).should("contain", "paper");
+    cy.get(".category-section").children().eq(2).should("contain", "baking");
+    cy.get(".category-section").children().eq(3).should("contain", "jewelry");
+    cy.get(".category-section").children().eq(4).should("contain", "scrapbook/floral");
+    cy.get(".category-section").children().eq(5).should("contain", "painting");
+    cy.get(".category-section").children().eq(6).should("contain", "drawing");
+    cy.get(".category-section").children().eq(7).should("contain", "wood working");
+    cy.get(".category-section").children().eq(8).should("contain", "pottery");
+    cy.get(".category-section").children().eq(9).should("contain", "other");
+
```
Comment (stephanie-roe) [inline cypress/e2e/1-getting-started/landingPage.cy.js:49]: very thorough testing- love this!!

File: src/Components/CraftCategories.js [hunk 2]
```diff
@@ -18,8 +19,9 @@ const CraftCategories = () => {
           <CategoryName>sewing/knitting</CategoryName>
         </CategoryBox>
         <CategoryBox
+          className={category === "Paper" ? "selected" : ""}
           style={{ background: colors.craftPurple }}
-          onClick={() => console.log("yeehaw")}
```
Comment (stephanie-roe) [inline src/Components/CraftCategories.js:22]: Thanks for cleaning up these console.logs

**Review by stephanie-roe — APPROVED** [review]
Looking great! I appreciate you doing such a thorough job for testing.

File: cypress/e2e/1-getting-started/landingPage.cy.js [hunk 3]
```diff
@@ -1,16 +1,111 @@
...
+  it("Should be ble to navigate through the Nav Bar buttons", () => {
+    cy.get(".nav-button").eq(1).click();
+    cy.url().should("contain", "http://localhost:3000/about");
+
+    cy.get(".nav-button").eq(0).click();
+    cy.url().should("contain", "http://localhost:3000");
+
+    cy.get(".nav-button").eq(2).click();
+    cy.url().should("contain", "http://localhost:3000/profile");
+
```
Comment (JoeRupp) [inline cypress/e2e/1-getting-started/landingPage.cy.js:108]: Nice tests! Covers a lot on the landing page and includes the nav bar

**Pull request merged**